.github/workflows/dev_deployment.yml

```diff
@@ -10,7 +10,7 @@ on:
 # permission can be added at job level or workflow level. Ref: https://github.com/aws-actions/configure-aws-credentials#OIDC
 permissions:
   id-token: write # This is required for requesting the JWT
-  contents: read # This is required for actions/checkout
+  contents: write # This is required for actions/checkout
```

Comment:
Since we need to update this permission to be "write", I feel more safe to do this in dev deployment pipeline instead of the prod one. So the data model generator will be trigger every time when we merge to main.

Reply:
Sounds good. 
The error seems to be  from the last step of `reusable_data_model_gen.yml`. Is it possible this `permissions: contents: write` be inside the `reusable_data_model_gen.yml` before `jobs:`?

Reply:
I tried that, but it returns me the error, I think if the dev_deployment config permission to be read, I can't do permission write in the data model gen yaml

Reply:
<img width="767" alt="Screen Shot 2024-08-22 at 11 27 31 AM" src="https://github.com/user-attachments/assets/2531280f-b25f-496b-8801-8da61ca0c83d">

Reply:
Is it because when you use ci_infrastructure.yml to tested it out, it already specifically set to `contents: read` but for dev_deployment.yml, this `content` has not been set yet so can be set in reusable yml jobs?

Reply:
dev_deployment also has content set to be read already

Reply:
I see, I must be looking at the wrong one that did not have it.
Thanks!